### DISCUSSION TITLE
How to handle a Person with a username and an optional real name

### DISCUSSION THREAD

**Discussion by karptonite** [schemaorg/schemaorg#3459] [Q&A]
I'm adding microdata for a forum where we have People who  all have a username, and also, optionally, a real name (family and given). I'm a bit stumped about what to use for `name`. Right now, I'm using the real name if they have one, and the username if they don't, but mixing data like that is pretty icky. Alternatively, I could always use `username` for the name, and then use `familyName` and `givenName` for the Real name, when available, but I don't know that saying that a person's `name` is their username is really the right approach either. Any thoughts?

**Comment by jvandriel** [comment]
I have been using `alternateName` for a `Person`'s username, due to the lack of a better alternative.

Having said that, there is a discussion going on about adding `UserAccount` (#3403).

**Reply by karptonite** [reply]
Thanks, that makes sense!
